pennylane/devices/default_qubit.py

```diff
@@ -587,7 +587,7 @@ def expval(self, observable, shot_range=None, bin_size=None):
                 # Coefficients and the state are not trainable, we can be more
                 # efficient in how we compute the Hamiltonian sparse matrix.
                 if observable.name == "Hamiltonian":
-                    Hmat = qml.utils.sparse_hamiltonian(observable, wires=self.wires)
+                    Hmat = observable.sparse_matrix(wire_order=self.wires)
                 elif observable.name == "SparseHamiltonian":
                     Hmat = observable.sparse_matrix()
```

Comment:
```suggestion
                if observable.name in {"Hamiltonian", "SparseHamiltonian"}:
                     Hmat = observable.sparse_matrix()
```
An optional simplification we can now make.